numpy/core/tests/test_multiarray.py

```diff
@@ -8997,6 +9013,24 @@ class Dummy: pass
         break_cycles()
         assert_(obj_ref() is None, "no references should remain")
 
+    def test_can_use_super(self):
+        class SuperFinalize(np.ndarray):
+            def __array_finalize__(self, obj):
+                self.saved_result = super().__array_finalize__(obj)
+
+        a = np.array(1).view(SuperFinalize)
+        assert_(a.saved_result is None)
+
+    def test_use_none_is_deprecated(self):
+        # Deprecated way that ndarray itself showed nothing needs finalizing.
+        class NoFinalize(np.ndarray):
+            __array_finalize__ = None
+
+        with pytest.warns(DeprecationWarning,
+                          match="__array_finalize__ = None"):
+            a = np.array(1).view(NoFinalize)
+        assert isinstance(a, NoFinalize)
```

Comment:
Could move to `test_deprecations` or add a test for when it is raised (the DeprecationTestCase always does this).  But not super important, the ref-counting is obviously correct and that is the easiest error.